### ISSUE TITLE
I can´t encrypt my journal

### ISSUE THREAD

**Issue by aisneto** [jrnl-org/jrnl#1832]
### Diagnostic output

jrnl: v4.1
Python: 3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0]
OS: Linux 6.1.0-13-amd64

### Current Behavior

Encrypting my journal file with "journal --encrypt" simply does not work. The process initiated by the command never finishes, and as a result, I do not receive any error messages or prompts to enter a password and save it on the keychain.

### Expected Behavior

Running the command prompts me to enter a password for encryption.

### Repro Steps

Simply run jrnl --encrypt

### Debug output

```
~$ jrnl --encrypt --debug 

[07:45:51] DEBUG    Logging start                                     main.py:32
           DEBUG    Parsed args:                                      main.py:42
                    Namespace(debug=True, preconfig_cmd=None,                   
                    postconfig_cmd=<function                                    
                    cmd_requires_valid_journal_name.<locals>.wrapper            
                    at 0x7f08d6bba980>, filename=None, template=None,           
                    on_date=None, today_in_history=False, month=None,           
                    day=None, year=None, start_date=None,                       
                    end_date=None, contains=None, strict=False,                 
                    starred=False, tagged=False, limit=None,                    
                    excluded=[], edit=False, delete=False,                      
                    change_time=None, export=False, tags=False,                 
                    short=False, config_override=[],                            
                    config_file_path='', text=[],                               
                    exclude_starred=False, exclude_tagged=False)                
           DEBUG    Reading configuration from file                install.py:91
                    /home/xxxx/.config/jrnl/jrnl.yaml                        
           DEBUG    Using configuration:                          install.py:117
                    "{                                                          
                        'colors': {                                             
                            'body': 'none',                                     
                            'date': 'black',                                    
                            'tags': 'yellow',                                   
                            'title': 'cyan'                                     
                        },                                                      
                        'default_hour': 9,                                      
                        'default_minute': 0,                                    
                        'editor': '',                                           
                        'encrypt': False,                                       
                        'highlight': True,                                      
                        'indent_character': '|',                                
                        'journals': {                                           
                            'default': {                                        
                                'encrypt': False,                               
                                'journal':                                      
                    '/home/aisneto/Documentos/Journal/journal.txt               
                    '                                                           
                            }                                                   
                        },                                                      
                        'linewrap': 79,                                         
                        'tagsymbols': '#@',                                     
                        'template': False,                                      
                        'timeformat': '%F %r',                                  
                        'version': 'v4.1'                                       
                    }"                                                          
           DEBUG    Using journal name: default                    config.py:206
           DEBUG    Updating configuration with specific journal   config.py:111
                    overrides:                                                  
                    {'encrypt': False, 'journal':                               
                    '/home/aisneto/Documentos/Journal/journal.txt'              
                    }                                                           
           DEBUG    Scoped config:                                 config.py:120
                    {                                                           
                        'colors': {                                             
                            'body': 'none',                                     
                            'date': 'black',                                    
                            'tags': 'yellow',                                   
                            'title': 'cyan'                                     
                        },                                                      
                        'default_hour': 9,                                      
                        'default_minute': 0,                                    
                        'editor': '',                                           
                        'encrypt': False,                                       
                        'highlight': True,                                      
                        'indent_character': '|',                                
                        'journals': {                                           
                            'default': {                                        
                                'encrypt': False,                               
                                'journal':                                      
                    '/home/aisneto/Documentos/Journal/journal.txt'              
                            }                                                   
                        },                                                      
                        'linewrap': 79,                                         
                        'tagsymbols': '#@',                                     
                        'template': False,                                      
                        'timeformat': '%F %r',                                  
                        'version': 'v4.1',                                      
                        'journal':                                              
                    '/home/aisneto/Documentos/Journal/journal.txt'              
                    }                                                           
           DEBUG    open_journal 'default'                        Journal.py:473
           DEBUG    start                                   BaseEncryption.py:16
           DEBUG    start                                     NoEncryption.py:12
           DEBUG    decrypting                              BaseEncryption.py:29
           DEBUG    decrypting                                NoEncryption.py:19
           DEBUG    opened Journal with 499 entries               Journal.py:136
           DEBUG    Clearing encryption method...                commands.py:111
           DEBUG    start                                   BaseEncryption.py:16
           DEBUG    start                           BasePasswordEncryption.py:19
           DEBUG    start                                 Jrnlv2Encryption.py:23
           DEBUG    encrypting                      BasePasswordEncryption.py:46
```

The proccess seems to be stuck on the last step "encrypting¨.

### Other Information

This is happening right after a fresh install. Could be due to missing packages, though I've already installed cryptography globally using apt.

**Comment by micahellison** [comment]
Hi @aisneto, thanks for filing this, and for all the detailed info.

I don't know what's going on exactly, but based on the lines in that debug log, it's probably a problem with one of these two things:
- The [keyring](https://pypi.org/project/keyring/) library on your machine. Even when you first encrypt, jrnl checks to see if you have a saved password for that journal name. Maybe there's some prompt that it's waiting for elsewhere in your environment. For instance, on some Mac configurations, I believe there's a prompt that pops up asking you to unlock your password manager, if you haven't already done so. That might make sense on a fresh machine. I don't see any [current issues](https://github.com/jaraco/keyring/issues) about it hanging unexpectedly. It sounds like you can disable the keyring for Python applications like jrnl by setting the environment variable `PYTHON_KEYRING_BACKEND=keyring.backends.null.Keyring` which might help narrow this issue down.
- A prompt from jrnl that's somehow being suppressed on your machine. Right after the keyring call, you should see a prompt that says: `Enter password for journal 'default':`. I'm not sure what would suppress this, though. We use rich's `input` method for this prompt and I don't see any [current issues](https://github.com/Textualize/rich/issues) about it being suppressed or hanging. I'm pretty sure the message goes to STDERR, which you are clearly already able to view since that's where those debug messages came from.

Otherwise, I don't think it's a missing package. [jrnl would crash with a ModuleNotFoundError](https://github.com/jrnl-org/jrnl/issues?q=is%3Aissue+ModuleNotFoundError) instead of hanging if that happened.

It might help others running into this issue to know how you installed jrnl. Was it pipx, brew, or something else?

**Comment by aisneto** [comment]
Thank you very much for your response! I did not think of any issues regarding the keyring, assuming that the keyring integration would come after the encryption. As you pointed out, that was the actual problem.

I installed my jrnl.sh using pipx. To resolve the issue, I simply installed keyring through pipx as well (pipx install keyring), and now I can decrypt my journal normally. Thank you! I should have tried that sooner.
